<!DOCTYPE html>
<html lang="en   ">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
    <title>OOMPA Packages</title>
    <link href="bianchi.css" rel="stylesheet" type="text/css" />
  </head>
<body>

<!-- <img src="tuba3.gif" alt="OOMPA logo" width="262px"  height="156px" /> -->
<div id="logobar">
  <img src="tuba2.gif" alt="OOMPA logo" height="262px" width="156px" />
<br />
<br />
<a href="index.html">Overview</a>
<br />
</div>
<div id="mainarea">
<h1>OOMPA</h1>
<h2>Object-Oriented Microarray and Proteomics Analysis</h2>
<h2>Package List (by Project)</h2>

<h3>OOMPA</h3>
<dl>
  <dt>oompaBase</dt>
  <dd>Class Unions, Matrix Operations, and Color Schemes for OOMPA</dd>
  <dt>oompaData</dt>
  <dd>Data to Illustrate OOMPA Algorithms</dd>
  <dt>PreProcess</dt>
  <dd>Basic Functions for Pre-Processing Microarrays</dd>
  <dt>ClassComparison</dt>
  <dd>Classes and Methods for "Class Comparison" Problems on Microarrays</dd>
  <dt>ClassDiscovery</dt>
  <dd>Classes and Methods for "Class Discovery" with Microarrays or Proteomics</dd>
  <dt>TailRank</dt>
  <dd>The Tail-Rank Statistic</dd>
  <dt>Polychrome</dt>
  <dd>Qualitative Palettes with Many Colors</dd>
  <dt>BetaModels</dt>
  <dd>Bayesian Analysis of Different Rates in Different Groups</dd>
</dl>

<h3>Class Prediction</h3>
<dl>
  <dt>BimodalIndex</dt>
  <dd>The Bimodality Index</dd>
  <dt>Modeler</dt>
  <dd>Classes and Methods for Training and Using Binary Prediction Models</dd>
  <dt>CrossValidate</dt>
  <dd>Classes and Methods for Cross Validation of "Class Prediction" with Microarrays or Proteomics </dd>
  <dt>GenAlgo</dt>
  <dd>Classes and Methods to Use Genetic Algorithms for Feature Selection</dd>
  <dt>Classprediction</dt>
  <dd>Classes and Methods for "Class Prediction" with Microarrays or Proteomics</dd>
</dl>

<h3>NameNeedle</h3>
<dl>
  <dt>NameNeedle</dt>
  <dd>Using Needleman-Wunsch to Match Sample Names</dd>
</dl>

<h3>SIBER</h3>
<dl>
  <dt>SIBER</dt>
  <dd>Systematic Identification of Bimodally Expressed Genes Using RNAseq Data</dd>
</dl>

<h3>integIRTy</h3>
<dl>
  <dt>integIRTy</dt>
  <dd>Integrating Multiple Modalities of High Throughput Assays Using
  Item Response Theory </dd>
</dl>

<h3>UMPIRE</h3>
<dl>
  <dt>UMPIRE</dt>
  <dd>The Ultimate Microrray Prediction, Reality and Inference
  Engine</dd>
</dl>

<h3>Thresher</h3>
<dl>
  <dt>PCDimension</dt>
  <dd>Finding the Number of Significant Principal Components</dd>
  <dt>Thresher</dt>
  <dd>Threshing and Reaping for Principal Components</dd>
</dl>

<h3>SuperCurve</h3>
<dl>
  <dt>SuperCurveSampleData</dt>
  <dd>'SuperCurve' RPPA Sample Data</dd>
  <dt>SuperCurve</dt>
  <dd>RPPA Analysis Package</dd>
  <dt>tclish</dt>
  <dd>'Tcl/Tk' Utilities Package</dd>
  <dt>SuperCurveGUI</dt>
  <dd>'SuperCurve' GUI Package</dd>
  <dt>SlideDesignerGUI</dt>
  <dd>Slide Designer GUI Package</dd>
</dl>


<BR />
<BR />
<!-- R-Forge Logo -->
<table border="0" width="100%" cellspacing="0" cellpadding="0">
<tr><td>
<a href="http://r-forge.r-project.org/">
 <img src="http://r-forge.r-project.org/themes/rforge//imagesrf/logo.png" 
      border="0" alt="R-Forge Logo" /> </a> </td> </tr>
</table>

</div> <!-- end mainarea div-->


</body>
</html>
