<html><head><title>The Bum Class</title>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<link rel="stylesheet" type="text/css" href="Rchm.css">
</head>
<body>

<table width="100%"><tr><td>Bum-class(ClassComparison)</td><td align="right">R Documentation</td></tr></table><object type="application/x-oleobject" classid="clsid:1e2a7bd0-dab9-11d0-b93a-00c04fc99f9e">
<param name="keyword" value="R:   Bum">
<param name="keyword" value="R:   Bum-class">
<param name="keyword" value="R:   countSignificant,Bum-method">
<param name="keyword" value="R:   cutoffSignificant,Bum-method">
<param name="keyword" value="R:   hist,Bum-method">
<param name="keyword" value="R:   image,Bum-method">
<param name="keyword" value="R:   selectSignificant,Bum-method">
<param name="keyword" value="R:   summary,Bum-method">
<param name="keyword" value="R:   likelihoodBum">
<param name="keyword" value=" The Bum Class">
</object>


<h2>The Bum Class</h2>


<h3>Description</h3>

<p>
The <code>Bum</code> class is used to fit a beta-uniform mixture model to a
set of p-values.
</p>


<h3>Usage</h3>

<pre>
Bum(pvals, ...)
## S4 method for signature 'Bum':
summary(object, tau=0.01, ...)
## S4 method for signature 'Bum':
hist(x, res=100, xlab='P Values', main='', ...)
## S4 method for signature 'Bum':
image(x, ...)
## S4 method for signature 'Bum':
cutoffSignificant(object, alpha, by='FDR', ...)
## S4 method for signature 'Bum':
selectSignificant(object, alpha, by='FDR', ...)
## S4 method for signature 'Bum':
countSignificant(object, alpha, by='FDR', ...)
likelihoodBum(object)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>pvals</code></td>
<td>
A numeric vector containing values between 0 and 1</td></tr>
<tr valign="top"><td><code>object</code></td>
<td>
A <code>Bum</code> object</td></tr>
<tr valign="top"><td><code>tau</code></td>
<td>
A real number between 0 and 1, representing a cutoff on the
p-values.</td></tr>
<tr valign="top"><td><code>x</code></td>
<td>
A <code>Bum</code> object</td></tr>
<tr valign="top"><td><code>res</code></td>
<td>
A positive integer; the resolution at which to plot the
fitted distribution curve.</td></tr>
<tr valign="top"><td><code>xlab</code></td>
<td>
Label for the x axis</td></tr>
<tr valign="top"><td><code>main</code></td>
<td>
Graph title</td></tr>
<tr valign="top"><td><code>alpha</code></td>
<td>
Either the false discovery rate (if <code>by = 'FDR'</code>) or
the posterior probability (if <code>by = 'EmpiricalBayes'</code>)</td></tr>
<tr valign="top"><td><code>by</code></td>
<td>
String denoting the method to use for determining
cutoffs. The chioces are 'FDR', 'FalseDiscovery', or
'EmpiricalBayes'. Since the test is implemented with
<code><a onclick="findlink('base', 'match.arg.html')" style="text-decoration: underline; color: blue; cursor: hand">match.arg</a></code>, unique abbreviations also work.</td></tr>
<tr valign="top"><td><code>...</code></td>
<td>
All methods are defined to accept additional arguments in
order to allow flexibility in designing derived classes.  The usual
graphical parameters can be supplied to <code>hist</code> and <code>image</code>.</td></tr>
</table>

<h3>Details</h3>

<p>
The BUM method was introduced by Stan Pounds and Steve Morris,
although it was simultaneously discovered by several other
researchers.  It is generally applicable to any analysis of microarray
or proteomics data that performs a separate statistical hypothesis
test for each gene or protein, where each test produces a p-value that
would be valid if the analyst were only performing one statistical
test.  When performing thousands of statistical tests, however, those
p-values no longer have the same interpretation as Type I error
rates.  The idea behind BUM is that, under the null hypothesis that
none of the genes or proteins is interesting, the expected
distribution of the set of p-values is uniform.  By contrast, if some
of the genes are interesting, then we should see an overabundance of
small p-values (or a spike in the histogram near zero).  We can model
the alternative hypothesis with a beta distribution, and view the set
of all p-values as a mixture distribution.
</p>
<p>
Fitting the BUM model is straightforward, using a nonlinear optimizer
to compute the maximum likelihood parameters.  After the model has
been fit, one can easily determine cutoffs on the p-values that
correspond to desired false discovery rates.  Alternatively, the
original Pounds and Morris paper shows that their results can be
reinterpreted to recover the empirical Bayes method introduced by
Efron and Tibshirani.  Thus, one can also determine cutoffs by
specifying a desired posterior probability of signficance.
</p>


<h3>Value</h3>

<p>
Graphical functions (<code>hist</code> and <code>image</code>) invisibly return the
object on which they were invoked.
<br>
The <code>cutoffSignficant</code> method returns a real number between zero
and one. P-values below this cutoff are considered statistically
significant at either the specified false discovery rate or at the
specified posterior probability.
<br>
The <code>selectSignficant</code> method returns a vector of logical values
whose length is equal to the length of the vector of p-values that was
used to construct the <code>Bum</code> object.  True values in the return
vector mark the statistically signficant p-values.
<br>
The <code>countSignificant</code> method returns an integer, the number of
statistically significant p-values.
<br>
The <code>summary</code> method returns an object of class
<code><a href="cc00-1-bumsum.html">BumSummary</a></code>.</p>

<h3>Creating Objects</h3>

<p>
Although objects can be created directly using <code>new</code>, the most
common usage will be to pass a vector of p-values to the
<code>Bum</code> function.
</p>


<h3>Slots</h3>

<dl>
<dt><code>pvals</code>:</dt><dd>The vector of p-values used to construct the object.</dd>
<dt><code>ahat</code>:</dt><dd>Model parameter</dd>
<dt><code>lhat</code>:</dt><dd>Model parameter</dd>
<dt><code>pihat</code>:</dt><dd>Model parameter</dd>
</dl>

<h3>Methods</h3>

<dl>
<dt>summary(object, tau=0.01, ...)</dt><dd>For each value of the p-value
cutoff <code>tau</code>, computes estimates of the fraction of true
positives (TP), false negatives (FN), false positives (FP), and
true negatives (TN).</dd>
<dt>hist(x, res=100, xlab='P Values', main='', ...)</dt><dd>Plots a
histogram of the object, and overlays (1) a straight line to indicate
the contribution of the uniform component and (2) the fitted
beta-uniform distribution from the observed values. Colors in the
plot are controlled by
<code><a onclick="findlink('PreProcess', 'semanticColors.html')" style="text-decoration: underline; color: blue; cursor: hand">COLOR.EXPECTED</a></code> and 
<code><a onclick="findlink('PreProcess', 'semanticColors.html')" style="text-decoration: underline; color: blue; cursor: hand">COLOR.OBSERVED</a></code>.
</dd>
<dt>image(x, ...)</dt><dd>Produces four plots in a 2x2 layout: (1) the
histogram produced by <code>hist</code>; (2) a plot of cutoffs against
the desired false discovery rate; (3) a plot of cutoffs against
the posterior probability of coming from the beta component; and
(4) an ROC curve.</dd>
<dt>cutoffSignificant(object, alpha, by='FDR', ...)</dt><dd>Computes the
cutoff needed for significance, which in this case means arising
from the beta component rarther than the uniform component of the
mixture. Significance is specified either by the false discovery
rate (when <code>by = 'FDR'</code> or <code>by = 'FalseDiscovery'</code>) or
by the posterior probability (when <code>by = 'EmpiricalBayes'</code>)</dd>
<dt>selectSignificant(object, alpha, by='FDR', ...)</dt><dd>Uses
<code>cutoffSignificant</code> to determine a logical vector that
indicates which of the p-values are significant.</dd>
<dt>countSignificant(object, alpha, by='FDR', ...)</dt><dd>Uses
<code>selectSignificant</code> to count the number of significant
p-values.</dd>
</dl>

<h3>Author(s)</h3>

<p>
Kevin R. Coombes &lt;kcoombes@mdanderson.org&gt;
</p>


<h3>References</h3>

<p>
Pounds S, Morris SW. Estimating the occurrence of false
positives and false negatives in microarray studies by approximating
and partitioning the empirical distribution of
p-values. Bioinformatics. 2003 Jul 1;19(10):1236-42. 
</p>
<p>
Benjamini Y, Hochberg Y.  Controlling the false discovery rate: a
practical and powerful approach to multiple testing. J Roy Statist Soc
B, 1995; 57: 289-300.
</p>
<p>
Efron B, Tibshirani R: Empirical bayes methods and false discovery
rates for microarrays. Genet Epidemiol 2002, 23: 70-86.
</p>


<h3>See Also</h3>

<p>
Two classes that produce lists of p-values that can (and often
should) be analyzed using BUM are <code><a href="cc01-0-multiT.html">MultiTtest</a></code> and
<code><a href="cc03-0-multiLM.html">MultiLinearModel</a></code>. Also see <code><a href="cc00-1-bumsum.html">BumSummary</a></code>.
</p>


<h3>Examples</h3>

<pre>
fake.data &lt;- c(runif(700), rbeta(300, 0.3, 1))
a &lt;- Bum(fake.data)
hist(a, res=200)

alpha &lt;- (1:25)/100
plot(alpha, cutoffSignificant(a, alpha, by='FDR'),
     xlab='Desired False Discovery Rate', type='l',
     main='FDR Control', ylab='Significant P Value')

GAMMA &lt;- 5*(10:19)/100
plot(GAMMA, cutoffSignificant(a, GAMMA, by='EmpiricalBayes'),
     ylab='Significant P Value', type='l',
     main='Empirical Bayes', xlab='Posterior Probability')

b &lt;- summary(a, (0:100)/100)
be &lt;- b@estimates
sens &lt;- be$TP/(be$TP+be$FN)
spec &lt;- be$TN/(be$TN+be$FP)
plot(1-spec, sens, type='l', xlim=c(0,1), ylim=c(0,1), main='ROC Curve')
points(1-spec, sens)
abline(0,1)

image(a)

countSignificant(a, 0.05, by='FDR')
countSignificant(a, 0.99, by='Emp')

#cleanup
rm(a, b, be, alpha, GAMMA, sens, spec, fake.data)
</pre>

<script Language="JScript">
function findlink(pkg, fn) {
var Y, link;
Y = location.href.lastIndexOf("\\") + 1;
link = location.href.substring(0, Y);
link = link + "../../" + pkg + "/chtml/" + pkg + ".chm::/" + fn;
location.href = link;
}
</script>


<hr><div align="center">[Package <em>ClassComparison</em> version 2.8.0 <a href="00Index.html">Index]</a></div>

</body></html>
