<html><head><title>The MultiWilcoxonTest Class</title>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<link rel="stylesheet" type="text/css" href="Rchm.css">
</head>
<body>

<table width="100%"><tr><td>MultiWilcoxonTest-class(ClassComparison)</td><td align="right">R Documentation</td></tr></table><object type="application/x-oleobject" classid="clsid:1e2a7bd0-dab9-11d0-b93a-00c04fc99f9e">
<param name="keyword" value="R:   MultiWilcoxonTest">
<param name="keyword" value="R:   MultiWilcoxonTest-class">
<param name="keyword" value="R:   countSignificant,MultiWilcoxonTest-method">
<param name="keyword" value="R:   cutoffSignificant,MultiWilcoxonTest-method">
<param name="keyword" value="R:   hist,MultiWilcoxonTest-method">
<param name="keyword" value="R:   plot,MultiWilcoxonTest,missing-method">
<param name="keyword" value="R:   selectSignificant,MultiWilcoxonTest-method">
<param name="keyword" value="R:   summary,MultiWilcoxonTest-method">
<param name="keyword" value=" The MultiWilcoxonTest Class">
</object>


<h2>The MultiWilcoxonTest Class</h2>


<h3>Description</h3>

<p>
The <code>MultiWilcoxonTest</code> class is used to perform
row-by-row Wilcoxon rank-sum tests on a data matrix.  Significance
cutoffs are determined by the empirical Bayes method of Efron and
Tibshirani
</p>


<h3>Usage</h3>

<pre>
MultiWilcoxonTest(data, classes, histsize = NULL)
## S4 method for signature 'MultiWilcoxonTest':
summary(object, prior=1, significance=0.9, ...)
## S4 method for signature 'MultiWilcoxonTest':
hist(x, xlab='Rank Sum',
 ylab='Prob(Different | Y)', main='', ...)
## S4 method for signature 'MultiWilcoxonTest, missing':
plot(x, prior=1, significance=0.9,
 ylim=c(-0.5, 1), xlab='Rank Sum', ylab='Prob(Different | Y)', ...)
## S4 method for signature 'MultiWilcoxonTest':
cutoffSignificant(object, prior, significance, ...)
## S4 method for signature 'MultiWilcoxonTest':
selectSignificant(object, prior, significance, ...)
## S4 method for signature 'MultiWilcoxonTest':
countSignificant(object, prior, significance, ...)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>data</code></td>
<td>
Either a data frame or matrix with numeric values or an
<code><a onclick="findlink('Biobase', 'class.ExpressionSet.html')" style="text-decoration: underline; color: blue; cursor: hand">ExpressionSet</a></code> as defined
in the BioConductor tools for analyzing microarray data.</td></tr>
<tr valign="top"><td><code>classes</code></td>
<td>
If <code>data</code> is a data frame or matrix, then classes
must be either a logical vector or a factor. If <code>data</code> is an
<code>ExpressionSet</code>, then <code>classes</code> can be a character string that
names one of the factor columns in the associated
<code><a onclick="findlink('Biobase', 'phenoData.html')" style="text-decoration: underline; color: blue; cursor: hand">phenoData</a></code> subobject.</td></tr>
<tr valign="top"><td><code>histsize</code></td>
<td>
An integer; the number of bins used for the histogram
summarizing the Wilcoxon statistics. When <code>NULL</code>, each discrete
rank-sum value gets its own bin.</td></tr>
<tr valign="top"><td><code>object</code></td>
<td>
an object of the <code>MultiWilcoxonTest</code> class.</td></tr>
<tr valign="top"><td><code>x</code></td>
<td>
an object of the <code>MultiWilcoxonTest</code> class.</td></tr>
<tr valign="top"><td><code>xlab</code></td>
<td>
Label for the x axis</td></tr>
<tr valign="top"><td><code>ylab</code></td>
<td>
Label for the y axis</td></tr>
<tr valign="top"><td><code>ylim</code></td>
<td>
Plotting limits on the y-axis</td></tr>
<tr valign="top"><td><code>main</code></td>
<td>
Graph title</td></tr>
<tr valign="top"><td><code>prior</code></td>
<td>
Prior probability that an arbitrary gene is not differentially
expressed, or that an arbitrary row does not yield a significant
Wilcoxon rank-sum statistic. </td></tr>
<tr valign="top"><td><code>significance</code></td>
<td>
Desired level of posterior probability</td></tr>
<tr valign="top"><td><code>...</code></td>
<td>
Additional graphical parameters.</td></tr>
</table>

<h3>Details</h3>

<p>
See the paper by Efron and Tibshirani.
</p>


<h3>Value</h3>

<p>
The standard methods <code>summary</code>, <code>hist</code>, and <code>plot</code>
return what you would expect.
<br>
The <code>cutoffSignificant</code> method returns a list of two
integers. Rank-sum values msaller than the first value or larger than
the second value are statistically significant in the sense that their
posterior probability exceeds the specified <code>significance</code> level
given the assumptions about the <code>prior</code> probability of not being
significant.
<br>
The <code>selectSignificant</code> method returns a vector of logical values
identifying the significant test results, and <code>countSignificant</code>
returns an integer counting the number of significant test results.</p>

<h3>Creating Objects</h3>

<p>
As usual, objects can be created by <code>new</code>, but better methods are
available in the form of the <code>MultiWilcoxonTest</code> function. The
inputs to this function are the same as those used for row-by-row
statistical tests throughout the ClassComparison package; a detailed
description can be found in the <code><a href="cc01-0-multiT.html">MultiTtest</a></code> class.
</p>
<p>
The constructor computes row-by-row Wilcoxon rank-sum statistics on
the input <code>data</code>, comparing the two groups defined by the
<code>classes</code> argument. It also estimates the observed and
theoretical (expected) density functions for the collection of
rank-sum statistics.
</p>
<p>
The additional input argument, <code>histsize</code> is usually best left to
its default value. In certain pathological cases, we have found it
necessary to use fewer bins; one suspects that the underlying model
does not adequately capture the complexity of those situations.
</p>


<h3>Slots</h3>

<dl>
<dt><code>rank.sum.statistics</code>:</dt><dd>A numeric vector containing the
computed rank-sum statistics.</dd>
<dt><code>xvals</code>:</dt><dd>A numeric vector, best thought of as the vector
of possible rank-sum statistics given the sizes of the two groups.</dd>
<dt><code>theoretical.pdf</code>:</dt><dd>A numeric vector containing the
theoretical density function evaluated at the points of
<code>xvals</code>.</dd>
<dt><code>pdf</code>:</dt><dd>A numeric vector containing the empirical density
function computed at the points of <code>xvals</code>.</dd>
<dt><code>unravel</code>:</dt><dd>A numeric vector containing a smoothed
estimate (by Poisson regression using B-splines) of the empirical
density function evaluated at <code>xvals</code>.</dd>
<dt><code>groups</code>:</dt><dd>A vector containing the names of the groups
defined by <code>classes</code>.</dd>
<dt><code>call</code>:</dt><dd>An object of class <code>call</code> representing the
function call that created the object. </dd>
</dl>

<h3>Methods</h3>

<dl>
<dt>summary(object, prior=1, significance=0.9, ...)</dt><dd>Write out a summary of the object. For a given value of the
<code>prior</code> probability of not being differentially expressed and
a given significance cutoff on the posterior probability, reports
the cutoffs and number of items in both tails of the distribution.</dd>
<dt>hist(x, xlab='Rank Sum', ylab='Prob(Different|Y)', main='', ...)</dt><dd>Plot a histogram of the rank-sum statistics, with overlaid curves
represnting the expected and observed distributions.  Colors of
the curves are controlled by
<code><a onclick="findlink('PreProcess', 'semanticColors.html')" style="text-decoration: underline; color: blue; cursor: hand">COLOR.EXPECTED</a></code> and 
<code><a onclick="findlink('PreProcess', 'semanticColors.html')" style="text-decoration: underline; color: blue; cursor: hand">COLOR.OBSERVED</a></code>.
</dd>
<dt>plot(x, prior=1, significance=0.9, ylim=c(-0.5, 1), xlab='Rank
Sum', ylab='Prob(Different | Y)', ...)</dt><dd>Plots the posterior probability of being differentially expressed
for given values of the <code>prior</code>.  Horixontal lines are added
at each specified <code>significance</code> level for the posterior
probability.
</dd> 
<dt>cutoffSignificant(object, prior, significance, ...)</dt><dd>Determine
cutoffs on the rank-sum statistic at the desired significance level.</dd>
<dt>selectSignificant(object, prior, significance, ...)</dt><dd>Compute a
logical vector for selecting significant test results.</dd>
<dt>countSignificant(object, prior, significance, ...)</dt><dd>Count the
number of significant test results.</dd>
</dl>

<h3>Author(s)</h3>

<p>
Kevin R. Coombes &lt;kcoombes@mdanderson.org&gt;
</p>


<h3>References</h3>

<p>
Efron B, Tibshirani R: Empirical bayes methods and false discovery
rates for microarrays. Genet Epidemiol 2002, 23: 70-86.
</p>
<p>
Pounds S, Morris SW. Estimating the occurrence of false
positives and false negatives in microarray studies by approximating
and partitioning the empirical distribution of
p-values. Bioinformatics. 2003 Jul 1;19(10):1236-42.
</p>


<h3>See Also</h3>

<p>
Implementation is handled in part by the functions <code><a href="cc02-1-dwil.html">dwil</a></code>
and <code><a href="cc02-2-rankSum.html">rankSum</a></code>.  The empirical Bayes results for
alternative tests (such as <code><a href="cc01-0-multiT.html">MultiTtest</a></code>) can be obtained
using the beta-uniform mixture model in the <code><a href="cc00-0-bum.html">Bum</a></code> class.
</p>


<h3>Examples</h3>

<pre>
ng &lt;- 10000
ns &lt;- 15
nd &lt;- 200
fake.class &lt;- factor(rep(c('A', 'B'), each=ns))
fake.data &lt;- matrix(rnorm(ng*ns*2), nrow=ng, ncol=2*ns)
fake.data[1:nd, 1:ns] &lt;- fake.data[1:nd, 1:ns] + 2
fake.data[(nd+1):(2*nd), 1:ns] &lt;- fake.data[(nd+1):(2*nd), 1:ns] - 2

a &lt;- MultiWilcoxonTest(fake.data, fake.class)
hist(a)
plot(a)
plot(a, prior=0.85)
abline(h=0)

cutoffSignificant(a, prior=0.85, signif=0.95)
countSignificant(a, prior=0.85, signif=0.95)

# cleanup
rm(ng, ns, nd, fake.class, fake.data, a)
</pre>

<script Language="JScript">
function findlink(pkg, fn) {
var Y, link;
Y = location.href.lastIndexOf("\\") + 1;
link = location.href.substring(0, Y);
link = link + "../../" + pkg + "/chtml/" + pkg + ".chm::/" + fn;
location.href = link;
}
</script>


<hr><div align="center">[Package <em>ClassComparison</em> version 2.8.0 <a href="00Index.html">Index]</a></div>

</body></html>
